### PR TITLE
Icon Library

### PR DESCRIPTION
**Pull request by ovbm**

### CODE REVIEW

**Comment by tpreusse** [comment]
Unfortunately the tree shaking does not work as hoped yet. Resulting in 1.56mb Icons:
Screenshot 2021-04-16 at 16 51 16

For reference, before we had 87kb:
Screenshot 2021-04-16 at 17 54 18

I'll look into modernizing styleguide module setup to fix this. But I think we have to pause it again for after some more important changes.

**Comment by ovbm** [comment]
works with react-icons/all-files in styleguide:

Screenshot 2021-04-16 at 19 05 17

**Comment by tpreusse** [comment]
Verified on my machine: I also get 110kb and once we use the icons via styleguide here we should get down to around 90kb like before. Make sure to also adapt the `IconContext` in `_app.js`—probably should be re-exported from the styleguide in order to remove all direct react-icons dependencies from the frontend.

**Review by tpreusse — COMMENTED** [review]
Unfortunately you're right, the publikator icons leak into here. Tree shaking is not where I hoped it'd be at by now.

The parsed size of the commons bundle goes up by 8% (from 665.29kb to 716.82kb). Long term we have to improve the styleguide module bundling in order to enable proper splitting and shaking.

Short term I see two options:
- `@project-r/styleguide/tooling` to export stuff we only need in Publikator/Admin
- import tooling icons directly from ``@react-icons/all-files` in Publikator/Admin

**Review by tpreusse — APPROVED** [review]
Before
```
+ First Load JS shared by all    454 kB
  ├ chunks/1240.a20b01.js        19.7 kB
  ├ chunks/252f366e.530515.js    3.34 kB
  ├ chunks/2726.9147dd.js        3.65 kB
  ├ chunks/2938.b68681.js        63 kB
  ├ chunks/3001.a97b17.js        17.5 kB
  ├ chunks/3408.00da8b.js        6.72 kB
  ├ chunks/3630.606f45.js        12 kB
  ├ chunks/3900.89ea13.js        45.9 kB
  ├ chunks/5628.47d0d9.js        4.83 kB
  ├ chunks/6472.e3770f.js        43.3 kB
  ├ chunks/6893.ca8fb5.js        21.5 kB
  ├ chunks/7151.f7ef2a.js        4.38 kB
  ├ chunks/8553.b6b76c.js        8.03 kB
  ├ chunks/95b64a6e.c743b9.js    1.91 kB
  ├ chunks/commons.fa0ea1.js     133 kB
  ├ chunks/framework.2934b5.js   47.1 kB
  ├ chunks/main.cb46d2.js        6.82 kB
  ├ chunks/pages/_app.d4e854.js  9.09 kB
  └ chunks/webpack.0d9af0.js     2.15 kB
```

after
```
+ First Load JS shared by all    460 kB
  ├ chunks/1240.a20b01.js        19.7 kB
  ├ chunks/1651.a4bfd8.js        21 kB
  ├ chunks/2726.9147dd.js        3.65 kB
  ├ chunks/2938.b68681.js        63 kB
  ├ chunks/3001.c3f527.js        17.2 kB
  ├ chunks/3408.2ca3bc.js        6.72 kB
  ├ chunks/3630.b553e3.js        11.9 kB
  ├ chunks/3900.89ea13.js        45.9 kB
  ├ chunks/5628.47d0d9.js        4.83 kB
  ├ chunks/6472.e3770f.js        43.3 kB
  ├ chunks/7151.403dea.js        4.39 kB
  ├ chunks/8181.6d4ef6.js        7.64 kB
  ├ chunks/commons.79fd70.js     146 kB
  ├ chunks/framework.2934b5.js   47.1 kB
  ├ chunks/main.cb46d2.js        6.82 kB
  ├ chunks/pages/_app.ec0c78.js  9.07 kB
  └ chunks/webpack.343f58.js     2.12 kB
```

Overall we gained 6kb which is totally fine for now (commons grew more but other chunks git lighter).

Let's do a round of testing in `.love` with the released styleguide.

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "translations-vote": "gsheets --key=1oBlTiPKIon4p5aek5vEXluWMn0TRwrk-JftznGjTmdQ --title=live --pretty --out components/Vote/translations-vote.json",
     "analyze": "ANALYZE=true next build",
     "sg:link": "./scripts/sg_link.sh",
+    "preinstall": "npm run sg:unlink",
```
Comment (tpreusse) [inline package.json:23]: make sure that get's removed again

Comment (tpreusse) [reply package.json:23]: (should happen automatically when installing the released styleguide)

**Pull request merged**